src/Detector/Python/PythonDetector.cs

```diff
@@ -24,61 +23,52 @@ public PythonDetector(ILogger<PythonDetector> logger)
         public PlatformDetectorResult Detect(DetectorContext context)
         {
             var sourceRepo = context.SourceRepo;
-            if (!sourceRepo.FileExists(PythonConstants.RequirementsFileName)
-                && !sourceRepo.FileExists(PythonConstants.SetupDotPyFileName))
-            {
-                _logger.LogDebug($"'{PythonConstants.SetupDotPyFileName}' or '{PythonConstants.RequirementsFileName}' " +
-                    $"does not exist in source repo");
-                return null;
-            }
-            else if (!sourceRepo.FileExists(PythonConstants.RequirementsFileName)
-                && sourceRepo.FileExists(PythonConstants.SetupDotPyFileName))
-            {
-                _logger.LogInformation($"'{PythonConstants.RequirementsFileName} doesn't exist in source repo.' " +
-                    $"Detected '{PythonConstants.SetupDotPyFileName}'that exists in source repo");
-            }
-            else
-            {
-                _logger.LogInformation($"'{PythonConstants.SetupDotPyFileName} doesn't exist in source repo.' " +
-                    $"Detected '{PythonConstants.RequirementsFileName}'that exists in source repo");
-            }
 
-            // This detects if a runtime.txt file exists if that is a python file
+            var isPythonApp = IsPythonApp(sourceRepo);
+
+            // This detects if a runtime.txt file exists and if that is a python file
             var versionFromRuntimeFile = DetectPythonVersionFromRuntimeFile(context.SourceRepo);
-            if (string.IsNullOrEmpty(versionFromRuntimeFile))
+            if (!isPythonApp && string.IsNullOrEmpty(versionFromRuntimeFile))
```

Comment:
Just want to double check if this logic should be: it's not a python app **or** nothing found in runtime file, then return null?

Reply:
No, the logic is fine...for example, you can imagine this line as following instead:
```
if (!isPythonApp)
{
    if (string.IsNullOrEmpty(versionFromRuntimeFile))
    {
        return null;
    }
}

```